lib/model.js

```diff
@@ -1412,6 +1412,13 @@ module.exports = (function() {
    * @return {Promise<undefined>}
    */
   Model.prototype.destroy = function(options) {
+    var self = this
+      , instances;
+
+    if (!options || !options.where) {
```

Comment:
I belive we support a `truncate` parameter, if that is set it should't throw an error.